feat: Added Authentication

mnath30 opened the pull request:
Added the following:
1. Login page
2. Signup page
3. Logout page
4. Added authentication slice and API calls for login and signup

mnath30 committed:
feat: Added Authentication

Added the following:
1. Login page
2. Signup page
3. Logout page
4. Added authentication slice and api calls for login and signup

shivanigangadharan commented on src/pages/Authentication/Login.jsx:
```diff
@@ -0,0 +1,111 @@
+              {showPassword ? (
+                <i
+                  className="fa-solid fa-eye"
+                  onClick={() => setShowPassword(false)}
```
Suggestion - Instead of having the onClick event on the icon, you can put it inside a `<span>` and place the onClick on that.

mnath30 replied:
Have added the span tag and moved the onclick to the span tag

shivanigangadharan commented on src/pages/UserPosts/Explore.jsx:
```diff
@@ -1,16 +1,20 @@
+      <div className="explore__container">
+        <div className="grid explore__grid">
+          {data.map((item, id) => (
+            <Posts post={item} key={id} />
```
Here I think you're giving the index as the key, which is anyway used as the default key. Better to use the post id for key.

mnath30 replied:
Currently used a dummy data for the UI part will make a note of this comment and incorporate it in the next pr

shivanigangadharan reviewed (commented):
Few suggestions :
- User can sign up by entering any random text in email field - it'll be good to have some validation there.
- Can have a `cursor: pointer` property in the eye icon for show password

Rest everything looks good, I also like that you have disabled the submit button until the form is entirely filled. Nice work. 👍🏼

mnath30 committed:
fix: Minor fixes

Added the following fixes:
1. Added email validation to signup page
2. Added the span tag for the onClick event instead of directly adding on click event to icon
3. Added pointer to the password visibility icons

mnath30 commented:
> Few suggestions :
> 
> * User can sign up by entering any random text in email field - it'll be good to have some validation there.
> * Can have a `cursor: pointer` property in the eye icon for show password
> 
> Rest everything looks good, I also like that you have disabled the submit button until the form is entirely filled. Nice work. 👍🏼

Thankyou so much Shivani for all your review comments
Have added the cursor pointer and the email validation
